app/models/search.rb

```diff
@@ -62,18 +62,13 @@ def query_cd_api
   end
 
   def query_cda
-    send(:"#{filter}_query")
-  end
-
-  def case_reference_query
-    CourtDataAdaptor::Query::ProsecutionCase.new(term)
-  end
-
-  def defendant_reference_query
-    CourtDataAdaptor::Query::Defendant::ByReference.new(term)
-  end
-
-  def defendant_name_query
-    CourtDataAdaptor::Query::Defendant::ByName.new(term, dob:)
+    case filter
+    when 'case_reference'
+      CourtDataAdaptor::Query::ProsecutionCase.new(term)
+    when 'defendant_reference'
+      CourtDataAdaptor::Query::Defendant::ByReference.new(term)
+    when 'defendant_name'
+      CourtDataAdaptor::Query::Defendant::ByName.new(term, dob:)
+    end
```

Comment:
This is to simplify `query_cda`. There is no needs of dynamic call `send(:"#{filter}_query")` plus three private methods.
A simple case switch does a great job.